tianshou/trainer/base.py

```diff
@@ -167,7 +167,7 @@ def __init__(
             save_best_fn = save_fn
 
         self.policy = policy
-        self.buffer = buffer
+        self.buffer = policy.process_buffer(buffer)
```

Comment:
```suggestion
        if buffer is not None:
             buffer = policy.process_buffer()
        self.buffer = buffer
```

Now that I think about it - does it make sense outside of offline learning?

Reply:
If you agree, I'd address this in a follow-up. Was thinking of adding a `_post_init` method to the trainer

Reply:
> Now that I think about it - does it make sense outside of offline learning?

I can think of a case If somebody wants to start with a previously collected buffer and wants to preprocess it before starting learning.